support-modules/rest/src/main/scala/com/gu/rest/WebServiceHelper.scala

```diff
@@ -172,13 +176,13 @@ trait WebServiceHelper[Error <: Throwable] {
   }
   // scalastyle:on line.size.limit
 
-  private def buildRequest(endpoint: String, headers: ParamMap, params: ParamMap): Request.Builder =
+  protected def buildRequest(endpoint: String, headers: ParamMap, params: ParamMap): Request.Builder =
     new Request.Builder()
       .url(endpointUrl(endpoint, params))
       .headers(buildHeaders(headers))
 
   private def endpointUrl(endpoint: String, params: ParamMap): HttpUrl = {
-    val withSegments = endpoint.split("/").foldLeft(urlBuilder) {
+    val withSegments = endpoint.split("/", -1).foldLeft(urlBuilder) {
```

Comment:
what is the -1 for?
```
scala> "asdf/asdas/errwr".split("/",-1)
val res0: Array[String] = Array(asdf, asdas, errwr)

scala> "asdf/asdas/errwr".split("/")
val res1: Array[String] = Array(asdf, asdas, errwr)

```

Reply:
ah I see gets rid of trailing slashes. Is that intended?
```
scala> "asdf/asdas/errwr/".split("/",-1)
val res2: Array[String] = Array(asdf, asdas, errwr, "")

scala> "asdf/asdas/errwr/".split("/")
val res3: Array[String] = Array(asdf, asdas, errwr)

```

Reply:
It means that we retain trailing slashes on urls when they're specified. Previously they were being stripped which isn't normally a problem, but the Zuora `batch-query/` endpoint is picky about it, it will 404 if the trailing slash isn't present

Reply:
got it, thanks, that makes complete sense 👍